node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1881,24 +1864,17 @@ async fn finalize_finality_signature_fetch<REv>(
 }
 
 /// Returns the EraId whose switch block should be used to obtain validator weights.
-fn get_era_id_for_validators_retrieval(
-    era_id: &EraId,
-    last_emergency_restart: Option<EraId>,
-) -> EraId {
-    // TODO: This function needs to handle multiple emergency restarts.
-    if *era_id != EraId::from(0) && last_emergency_restart != Some(*era_id) {
-        // For eras > 0 which are not the last emergency restart eras we need to use validator set
-        // from the previous era
+fn get_era_id_for_validators_retrieval(era_id: &EraId) -> EraId {
+    // TODO: This function needs to handle upgrades with changes to the validator set.
+    if *era_id != EraId::from(0) {
```

Comment:
Doesn't Clippy complain about using `if !=`, and that you should switch the `if` and `else` branches so it's `if ==`?
Anyway, you might as well just use `saturating_sub` now.